src/napkin_printer.ml

```diff
@@ -2365,6 +2365,70 @@ and printExpressionWithComments expr cmtTbl =
   let doc = printExpression expr cmtTbl in
   printComments doc cmtTbl expr.Parsetree.pexp_loc
 
+and printIfChain pexp_attributes ifs elseExpr cmtTbl =
+  let ifDocs = Doc.join ~sep:Doc.space (
+    List.mapi (fun i (ifExpr, thenExpr) ->
+      let ifTxt = if i > 0 then Doc.text "else if " else  Doc.text "if " in
+      match ifExpr with
+        | ParsetreeViewer.If ifExpr ->
+          let condition =
+            if ParsetreeViewer.isBlockExpr ifExpr then
+              printExpressionBlock ~braces:true ifExpr cmtTbl
+            else
+              let doc = printExpressionWithComments ifExpr cmtTbl in
+              match Parens.expr ifExpr with
+              | Parens.Parenthesized -> addParens doc
+              | Braced braces -> printBraces doc ifExpr braces
+              | Nothing -> Doc.ifBreaks (addParens doc) doc
+          in
+          Doc.concat [
+            ifTxt;
+            Doc.group (condition);
+            Doc.space;
+            let thenExpr = match ParsetreeViewer.processBracesAttr thenExpr with
+            (* This case only happens when coming from Reason, we strip braces *)
+            | (Some _, expr) -> expr
+            | _ -> thenExpr
+            in
+            printExpressionBlock ~braces:true thenExpr cmtTbl;
+          ]
+        | IfLet (pattern, conditionExpr, guard) ->
+          let patternDoc = printPattern pattern cmtTbl in
+          let conditionDoc = printExpressionWithComments conditionExpr cmtTbl in
+          let guardDocs = match guard with
+          | Some expr -> Doc.concat [
+            Doc.text "when ";
+            printExpressionWithComments expr cmtTbl;
+          ]
+          | None -> Doc.nil
+          in
+          Doc.concat [
```

Comment:
Can you add an extra test case for what happens with `if let Some(suuuuuuuuuuuuuuuuuuuuuuuuuperLongName) = anotherSuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName() {` ?

One possible option for a layout is having a newline after the `if let Some(x) =` when things exceed the line width.
```ocaml
Doc.group ( (* make this a separate group *)
  Doc.concat [ (* list of things *)
    lhs;
    Doc.text " =";
    Doc.indent ( (* increase the indentation *)
      Doc.concat [ (* list of things *)
        Doc.line; (* if the layout fits, make this a space, if it doesn't fit, make it a newline *)
        rhs
      ]
    )
  ]
)
```

Reply:
Can we maybe also add a test case for a record destructure that exceeds the line width:
```
if let {
  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName,
  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName2,
  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName3,
} = buildMyRecord() {
```

Reply:
Feel free to ping me if you need help.

Reply:
Added tests. The grouping is already ideal IMO because the pattern and the expression already break themselves down if they don't fit